packages/checks/src/is-hex-color-string.ts

```diff
@@ -0,0 +1,4 @@
+import { isString } from "util";
+
+export const isHexColorString = (x: any): x is string =>
+    isString(x) && /#([a-f0-9]{3}|[a-f0-9]{4}(?:[a-f0-9]{2}){0,2})\b/gi.test(x);
```

Comment:
This regexp is actually not really watertight:

```js
/#([a-f0-9]{3}|[a-f0-9]{4}(?:[a-f0-9]{2}){0,2})\b/gi.test("hi #fff")
// true

/#([a-f0-9]{3}|[a-f0-9]{4}(?:[a-f0-9]{2}){0,2})\b/gi.test("#fff hi")
//true
```

Instead I'd propose using line start & end conditions to force the full string is an hex color:

```js
/^#([a-f0-9]{3}|[a-f0-9]{4}(?:[a-f0-9]{2}){0,2})$/i.test("hi #fff")
// false
```

Lastly, please pre-declare the regexp as module local const outside the function for a ~15-20% speed up... See [`isUUID`](https://github.com/thi-ng/umbrella/blob/master/packages/checks/src/is-uuid.ts) for ref

Reply:
👍